packages/shylu/shylu_node/basker/example/Solve.cpp

```diff
@@ -32,24 +27,20 @@ int main(int argc, char* argv[])
   typedef Teuchos::RCP<Teuchos::ParameterList>       PList;
 
 
-  Teuchos::GlobalMPISession mpisession(&argc,&argv,&std::cout);
+  Tpetra::ScopeGuard mpisession(&argc,&argv);
 
-  Kokkos::initialize(argc, argv);
```

Comment:
No need for Kokkos::initialize ?

Reply:
ScopeGuard does the initilization of MPI and Kokkos.

Reply:
Tpetra::ScopeGuard takes its name from Kokkos::ScopeGuard.  The intent is for Tpetra to look and act as much like Kokkos as possible.  Tpetra::ScopeGuard only initializes and finalizes MPI and Kokkos if the user has not already done so.